src/Process/Builder/ProcessBuilder.php

```diff
@@ -49,24 +50,25 @@ public function getProcessForInitialTestRun(
         bool $skipCoverage,
         array $phpExtraOptions = []
     ): Process {
-        $includeArgs = PHP_SAPI == 'phpdbg' || $skipCoverage;
+        $includeArgs = PHP_SAPI == 'phpdbg';
 
-        $process = new Process(
+        // If we're expecting to receive a code coverage, test process must run in a vanilla environment
+        $processType = $skipCoverage ? Process::class : PhpProcess::class;
+
+        $process = new $processType(
             $this->testFrameworkAdapter->getExecutableCommandLine(
                 $this->testFrameworkAdapter->buildInitialConfigFile(),
                 $testFrameworkExtraOptions,
                 $includeArgs,
                 $phpExtraOptions
             ),
             null,
-            $includeArgs ? array_replace($_ENV, $_SERVER) : null,
+            null,
             null,
             null
```

Comment:
Instead off `null, null, ...` what about skiping those and doing a `$process->setTimeout(null)` afterwards?

I'm not sure which one is cleaner, what do you think?

Reply:
Yes, that's definitely better than `null, null, null, null`